File: src/gerador/gerarSenha.dart
import 'dart:math';

void main() {
  int comprimentoSenha = 12;
  String senhaGerada = gerarSenha(comprimentoSenha);
  print('Senha gerada: $senhaGerada');
}

String gerarSenha(int comprimento) {
  String caracteres =
      'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#\$%^&*()-_=+';
  Random random = Random();

  StringBuffer senha = StringBuffer();
  for (int i = 0; i < comprimento; i++) {
    int indice = random.nextInt(caracteres.length);
    senha.write(caracteres[indice]);
  }

  return senha.toString();
}

File: src/gerador/gerarSenha.js
function gerarSenha(comprimento) {
  const caracteres =
    'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()-_=+';
  let senha = '';

  for (let i = 0; i < comprimento; i++) {
    const indice = Math.floor(Math.random() * caracteres.length);
    senha += caracteres.charAt(indice);
  }

  return senha;
}

const comprimentoSenha = 12;
const senhaGerada = gerarSenha(comprimentoSenha);
console.log('Senha gerada: ' + senhaGerada);
